lib/Arch/Arch.cpp

```diff
@@ -803,10 +795,7 @@ void Arch::InitFromSemanticsModule(llvm::Module *module) const {
   const auto &dl = module->getDataLayout();
   const auto basic_block = module->getFunction("__remill_jump");
   CHECK_NOTNULL(basic_block);
-  const auto state_ptr_type =
-      NthArgument(basic_block, kStatePointerArgNum)->getType();
-  const auto state_type =
-      llvm::dyn_cast<llvm::StructType>(state_ptr_type->getPointerElementType());
+  const auto state_type = llvm::StructType::getTypeByName(module->getContext(), "struct.State");
```

Comment:
I'm trying to figure out how this should look. In `X86/Runtime/BasicBlock.cpp` I see some declarations for registers marked with `extern` but I don't see where they're instantiated. If I do `module->getGlobalVariable(...)` for those register names, they don't seem to be accessible.

Reply:
So you'd declare these global variables, mark them as used, then look at `__remill_intrinsics` and try to mark them used there too. The purpose is just that these variables exist as kind of "anchors" providing us a dependable way of accessing these structure types (as the global variable value type). These variables will otherwise never be defined/used, only declared.

Those variables that you're seeing are debugging registers that are global, whereas a `State` is "thread local." It's up to a runtime (code that links against lifted bitcode) to define those register global variables.

Reply:
[f29e920](https://github.com/lifting-bits/remill/pull/596/commits/f29e920a128d0cf2d4386b5e73de20bbd119c4cd)

Reply:
[20bc423](https://github.com/lifting-bits/remill/pull/596/commits/20bc423b7a170443ef2e85586a23c44524b20caa)